### PR TITLE
[SI-1091] Delete pending cats on Offender Release event

### PR DESCRIPTION
**Pull request by a6software**
Listen for Offender Release events. Delete any pending (not APPROVED or CANCELLED) cats, recats, and lite cats for the offender.

This is currently behind an inactive feature flag, and for the moment will only log what we would have deleted.

### CODE REVIEW

File: server/services/sqsService.js [hunk 1]
```diff
@@ -102,9 +103,19 @@ module.exports = function createSqsService(offenderService, formService) {
...
+            logger.info(
+              { event },
+              `${events.EVENT_DOMAIN_PRISONER_OFFENDER_SEARCH_PRISONER_RELEASED}: Received payload`
+            )
+            const nomsNumber = event?.additionalInformation?.nomsNumber
+            if (!nomsNumber) {
+              logger.warn({ MessageId: message.MessageId }, 'MessageId was missing a nomsNumber')
+              break
+            }
+            await formService.deletePendingCategorisations(nomsNumber, transactionalDbClient)
```
Comment (a6software) [inline server/services/sqsService.js:116]: INCORRECT - only do this if 'reason' is RELEASED

Comment (a6software) [reply server/services/sqsService.js:121]: fixed

File: server/services/sqsService.js [hunk 2]
```diff
@@ -102,9 +103,24 @@ module.exports = function createSqsService(offenderService, formService) {
...
+              `${events.EVENT_DOMAIN_PRISONER_OFFENDER_SEARCH_PRISONER_RELEASED}: Received payload`
+            )
+            const nomsNumber = event?.additionalInformation?.nomsNumber
+            if (!nomsNumber) {
+              logger.warn({ MessageId: message.MessageId }, 'MessageId was missing a nomsNumber')
+              break
+            }
+            const reason = event?.additionalInformation?.reason
+            if (!reason || reason !== 'RELEASED') {
+              logger.warn({ MessageId: message.MessageId, nomsNumber }, 'Reason was not "RELEASED"')
```
Comment (timothy-ness) [inline server/services/sqsService.js:118]: Should this be a warning? Is it something which could happen often or a kind of error?

**Review by timothy-ness — DISMISSED** [review]

File: server/services/sqsService.js (hunk 2 above)
Comment (a6software) [reply server/services/sqsService.js]: good spot, thanks - no, lazy copy paste from the nomsNumber check above, fixed (set to info) and pushed

**Review by timothy-ness — APPROVED** [review]

**Pull request merged**